modules/admin-ui/src/main/resources/OSGI-INF/event_endpoint.xml

```diff
@@ -52,5 +52,5 @@
              interface="org.opencastproject.security.urlsigning.service.UrlSigningService"
              bind="setUrlSigningService"/>
   <reference name="asset-manager" interface="org.opencastproject.assetmanager.api.AssetManager"
-             cardinality="1..1" policy="dynamic" bind="setAssetManager"/>
+             cardinality="1..1" policy="dynamic" bind="setAssetManager" unbind="unsetAssetManager"/>
```

Comment:
Not against your patch… but why on earth do we have a dynamic binding for this? I'm pretty sure the service will not work without the asset manager being present. And that's probably a question for all the other dynamic asset manager bindings as well. It seems like someone started and everyone else copied that XML…

Reply:
Is there a difference in behavior between policy dynamic or static, if the cardinality is 1..1? If the asset manager unbinds, the event endpoint service will be deactivated.

Reply:
` If the cardinality constraints cannot be maintained after a component configuration is activated, that is the reference becomes unsatisfied, the component configuration must be deactivated.`

http://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#service.component-reference.cardinality

Reply:
You are right, I missed the cardinality declaration. Still, there is a slight difference between `static, 1..1` and `dynamic, 1..1`. The latter lets you replace bound services at runtime, while the first would require a restart of the service. Dynamic may thus break if you do something with the specific service during activation. I don't think that's relevant for Opencast in most cases. In general, I would still prefer static, since that is simpler (less configuration, no unbind, …) and gives you some guarantees due to the guaranteed component restart. Of course, unless you implement something like a plugin system where you are actually often switching references and restarting the components every time would become costly.